text/0227-separate-semantic-conventions.md

```diff
@@ -0,0 +1,155 @@
+    to live. This directory will be hosted at
+    `https://opentelemetry.io/semantic_conventions/schemas/`
+
+Existing semantic conventions in the specification would be marked as
+deprecated, with documentation denoting the move.
+
+Additionally, if the semantic conventions eventually move to domain-specific
+director structure (e.g. `docs/{domain}/README.md`, with trace, metrics, events
+in the same file), then this can be refactored in the new repository, preserving
+git history.
+
+There will also be the following exceptions in the specification:
+
+- Semantic convetions used to implement API/SDK details will be fully specified
+  and will not be allowed to change in the Semantic Convention directory.
+  - Error/Exception handling will remain in the specification.
+  - SDK configuration interaction w/ semantic convention will remain in the
+    specification. Specifically `service.name`.
+- The specification may elevate some semantic conventions as necessary for
+  compatibility requirements, e.g. `service.instance.id` and
+  [Prometheus Compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md).
+
+These exceptions exist because:
+
+- Stable portions of the specification already rely on these conventions
+- These conventions are required to implement an SDK today.
+
+As such, the Specification should define the absolute minimum of reserved or
+required attribute names and their interaction with the SDK.
+
+## Internal details
+
+The following process would be used to ensure semantic conventions are
+seamlessly moved to their new location.
+
+- A new repository `opentelemetry/semantic_conventions` will be constructed with
+  the proposed format and necessary `Makefile` / tooling.
+- A moratorium will be placed on Semantic Convention PRs to the specififcation
+  repository. (Caveat that PRs related to this proposal would be allowed).
+- All semantic conventions in the Specification will be copied to the new
+  repository.
+- Semantic conventions in the Specification will be marked as deprecated with
+  links to the new location.
+- The Specification will be updated to require "special" conventions, like
+  `service.name` and configuration interaction.
+- Instrumentation authors will update their `SchemaURL` to
+   `https://opentelemetry.io/semantic_conventions/schemas/{semconv_version}`
+   from
+   `https://opentelemetry.io/schemas/{spec_version}`
+
+## Trade-offs and mitigations
+
+This proposal has a few drawbacks:
+
+- The semantic conventions will no longer be easily referencable form the specification.
+  - This is actually a benefit. We can ensure isolation of convention from
+    specification and require the Specification to use firm language for
+    attributes it requires, like `service.name`.
+  - We will provide x-links from existing location to the new location.
+- Semantic Convention version will no longer match the specification version.
```

Comment:
Will this be a new version starting at `v0.0.0` or will it carry on from the version at which the spec and conventions diverged?  The Go API and Collector both use the schema version in package identifiers for semantic convention constants.

Reply:
This will be a new version number.

I'd suggest you'll want a new package name to avoid breakage as well.  This is why we plan to move the entire SchemaURL.  Effectively this will look/feel like a "new" thing to existing users to migrate to, hopefully alleviating breakages.

Reply:
Changing the schema URL isn't an issue as there is a [constant](https://github.com/open-telemetry/opentelemetry-go/blob/2e54fbb3fede5b54f316b3a08eab236febd854e0/semconv/v1.18.0/schema.go#L20) for the URL in each semconv package.  The [package name](https://github.com/open-telemetry/opentelemetry-go/blob/v1.14.0/semconv/v1.18.0/doc.go#L20) we're using is `semconv` but the import paths include both `semconv` and the [corresponding spec versions](https://pkg.go.dev/go.opentelemetry.io/otel@v1.14.0/semconv).

What are other SIGs doing to keep versioned semconv constants available for use and how will this impact them?

Reply:
I disagree that we need to reset the version numbers and introduce a new package names. See my other comments.

Reply:
@tigrannajaryan How would you ensure "safe" moving of semconv from current specification to the new repo?  I do *not* want a stop-the-world migration for all SiGs.

Reply:
I posted a suggestion [here](https://github.com/open-telemetry/oteps/pull/227#discussion_r1168993110), let's continue in that thread.

Reply:
Updated this to include first released version.  PTAL.